### PR TITLE
[explore view] fix long query issue from Run in SQL LAB Button

### PR DESCRIPTION
**Pull request by graceguo-supercat**
### CATEGORY

Choose one

- [x] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
In Explore view, Superset allow user to open current query in SQL Lab:
Screen Shot 2020-03-21 at 4 20 31 PM

Current Superset will send a **GET** request with sql query. But when sql query is very complicate, it might be over 4000 characters, and out of the limit of browser/nginx limit. It will display 504 Bad Gateway error message.

This PR is to handle this request by submitting a hidden form with POST method.

### TEST PLAN
1. open explore view,
2. Click Run in SQL Lab button from dropdown menu
3. expect to see query in Sql Lab's query editor.

### REVIEWERS
@ktmud

### CODE REVIEW

**Review by ktmud — CHANGES_REQUESTED** [review]
Code looks good, but unfortunately it doesn't seem to work. The SQL Lab page needs to pick up the post data as well. Try this diff: https://github.com/apache/incubator-superset/commit/cefedf6dffac53924893d7765364c47b65d6fa74

File: superset/views/core.py [hunk 1]
```diff
@@ -2752,7 +2752,7 @@ def _get_sqllab_payload(user_id: int) -> Dict[str, Any]:
         }
 
     @has_access
-    @expose("/sqllab")
+    @expose("/sqllab", methods=["GET", "POST"])
```
Comment (graceguo-supercat) [inline superset/views/core.py]: this is to make /sqllab page accept POST

Comment (ktmud) [reply superset/views/core.py]: The request is accepted but the post data is not used, hence the SQL query is actually not pre-populated in the SQL editor as it is now. Previously the search query is acquired at the client side: https://github.com/apache/incubator-superset/commit/cefedf6dffac53924893d7765364c47b65d6fa74#diff-2782bdc5cd0627eef4679c85a44d4ac1L102

Comment (graceguo-supercat) [reply superset/views/core.py]: i merged your code snippet. now it works. thanks!

File: superset/views/core.py [hunk 2]
```diff
@@ -2743,19 +2743,30 @@ def _get_sqllab_payload(user_id: int) -> Dict[str, Any]:
...
+            "common": common_bootstrap_payload(),
+        }
+
+        tabs_data = self._get_sqllab_tabs(g.user.get_id())
+        payload.update(tabs_data)
+
+        form_data = request.form.get("form_data")
+        if form_data:
+            try:
+                payload["requested_query"] = json.loads(form_data)
```
Comment (graceguo-supercat) [inline superset/views/core.py]: i prefer call it request_query not form_data.

File: superset/views/core.py [hunk 3]
```diff
@@ -2743,19 +2743,30 @@ def _get_sqllab_payload(user_id: int) -> Dict[str, Any]:
...
     def sqllab(self):
         """SQL Editor"""
-        payload = self._get_sqllab_payload(g.user.get_id())
+        payload = {
+            "defaultDbId": config["SQLLAB_DEFAULT_DBID"],
+            "common": common_bootstrap_payload(),
+        }
+
+        tabs_data = self._get_sqllab_tabs(g.user.get_id())
+        payload.update(tabs_data)
```
Comment (ktmud) [inline superset/views/core.py]: Can rewrite with dict unpacking:

```python
        payload = {
            "defaultDbId": config["SQLLAB_DEFAULT_DBID"],
            "common": common_bootstrap_payload(),
            **self._get_sqllab_tabs(g.user.get_id())
        }
```

**Review by ktmud — APPROVED** [review]
Lgtm, with one style nit

**Pull request merged**